### PR TITLE
feat: add docker container example envbox template

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR adds an example template for running an `envbox` workspace as a Docker container. right now, our examples only include a Kubernetes-based template.

### CODE REVIEW

**Comment by sreya** [comment]
Should we really be recommending this? If people are going to be running sysbox containers directly on a VM shouldn't we encourage them to run the daemon with the `sysbox-runc` runtime like we do for dogfood instead of adding an additional layer of abstraction?

The reason envbox exists for K8s is because it's a bit more difficult to manage sysbox on the nodes.

**Comment by ericpaulsen** [comment]
@sreya I agree the daemon route should be the recommendation. though my thinking is, this template is helpful for quickly spinning up a Docker-enabled workspace, with no install necessary on the host. also, some customers are quite limited on what can be installed on the VM.

File: examples/templates/envbox-docker/main.tf [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    trap '[ $? -ne 0 ] && echo === Agent script exited with non-zero code. Sleeping infinitely to preserve #logs... && sleep infinity' EXIT
+    ${replace(coder_agent.dev.init_script, "/localhost|127\\.0\\.0\\.1/", "host.docker.internal")}
+    EOT
+  ]
+  env = [
+    "CODER_AGENT_TOKEN=${coder_agent.dev.token}",
+    "CODER_AGENT_URL=${data.coder_workspace.me.access_url}",
+    "CODER_INNER_IMAGE=${data.coder_parameter.image.value}",
+    "CODER_INNER_USERNAME=coder",
+    "CODER_BOOTSTRAP_SCRIPT=${coder_agent.dev.init_script}",
```
Comment (matifali) [inline examples/templates/envbox-docker/main.tf:107]: I could not found this documented in README.md or https://github.com/coder/envbox

File: examples/templates/envbox-docker/main.tf [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    "CODER_MOUNTS=/home/coder:/home/coder",
+    "CODER_ADD_FUSE=false",
+    "CODER_INNER_HOSTNAME=${data.coder_workspace.me.name}",
+    "CODER_ADD_TUN=false",
+    "CODER_CPUS=1",
+    "CODER_MEMORY=2"
```
Comment (matifali) [inline examples/templates/envbox-docker/main.tf:113]: Should we move these to `coder_parameter` data sources?

File: examples/templates/envbox-docker/main.tf [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  env = [
+    "CODER_AGENT_TOKEN=${coder_agent.dev.token}",
+    "CODER_AGENT_URL=${data.coder_workspace.me.access_url}",
+    "CODER_INNER_IMAGE=${data.coder_parameter.image.value}",
+    "CODER_INNER_USERNAME=coder",
+    "CODER_BOOTSTRAP_SCRIPT=${coder_agent.dev.init_script}",
+    "CODER_MOUNTS=/home/coder:/home/coder",
+    "CODER_ADD_FUSE=false",
+    "CODER_INNER_HOSTNAME=${data.coder_workspace.me.name}",
+    "CODER_ADD_TUN=false",
```
Comment (matifali) [inline examples/templates/envbox-docker/main.tf:111]: Is it required? Doesn't it default to `false`?

File: examples/templates/envbox-docker/main.tf [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  mounts {
+    target = "/var/lib/containers"
+    source = docker_volume.coder_volume.name
+    type   = "volume"
+  }
+  mounts {
+    target = "/var/lib/docker"
+    source = docker_volume.coder_volume.name
+    type   = "volume"
+  }
```
Comment (matifali) [inline examples/templates/envbox-docker/main.tf:162]: Why do we need these mounts and volumes?

File: examples/templates/envbox-k8s/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# envbox
+
+## Introduction
+
+`envbox` is an image that enables creating non-privileged containers capable of running system-level software (e.g. `dockerd`, `systemd`, etc) in Kubernetes.
```
Comment (matifali) [inline examples/templates/envbox-k8s/README.md:5]:
```suggestion
[`envbox`](https://github.com/coder/envbox) is an image that enables creating non-privileged containers capable of running system-level software (e.g. `dockerd`, `systemd`, etc) in Kubernetes and Docker.
```

File: examples/templates/envbox-k8s/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+| `CODER_CPUS`               | Dictates the number of CPUs to allocate the inner container. It is recommended to set this using the Kubernetes [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-container-fields-as-values-for-environment-variables).                                                                                                                                                                                 | false    |
+| `CODER_MEMORY`             | Dictates the max memory (in bytes) to allocate the inner container. It is recommended to set this using the Kubernetes [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-container-fields-as-values-for-environment-variables).                                                                                                                                                                          | false    |
+
```
Comment (matifali) [inline examples/templates/envbox-k8s/README.md:29]: `CODER_BOOTSTRAP_SCRIPT` is missing from this list. Also, I think we should remove all ENVs not used in this example and instead refer to https://github.com/coder/envbox for complete documentation.

File: examples/templates/envbox-k8s/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+it may be necessary to migrate existing envbox templates on newer versions of
+Coder. Consult the [migration](https://coder.com/docs/v2/latest/templates/parameters#migration)
+documentation for details on how to do so.
+
+To supply values to existing existing Terraform variables you can specify the
+`-V` flag. For example
+
+```bash
+coder templates create envbox --var namespace="mynamespace" --var max_cpus=2 --var min_cpus=1 --var max_memory=4 --var min_memory=1
+```
```
Comment (matifali) [inline examples/templates/envbox-k8s/README.md:42]: We can remove this section as we no longer support the migration flow. Also, the migration docs have been removed.

File: examples/templates/envbox-docker/README.md
Comment (matifali) [inline examples/templates/envbox-docker/README.md:1]: All of the comments on envbox-k8s README apply here, too.

**Review by matifali — CHANGES_REQUESTED** [review]

**Pull request closed without merging**